### DISCUSSION TITLE
ACR task problem z firewallem

### DISCUSSION THREAD

**Discussion by cybul7** [AzureCommunityPL/group#408] [Pomoc]
Cześć.
Chce zrobić taska, który będzie usuwał obrazy w ACR.
Dodałem agent pool oraz credentiale do wykonywania tego taska.
Agent pool działa w vnecie, który jest oczywiście w tym samym regionie co ACR.
Na acr ustawiony jest firewall (selected networks), w tasku dostaje błąd:

2022/03/07 13:58:57 Logging in to registry: XXXXXXXXX.azurecr.io failed to login, ran out of retries: failed to set docker credentials: Error response from daemon: Get "https://XXXXXXX.io/v2/": denied: client with IP '52.178.20.51' is not allowed access. Refer https://aka.ms/acr/firewall to grant access. : exit status 1

Agent dostaje losowy adres IP, więc nie mogę dodać go do wyjątku w firewallu w ACR.
Nie mogę też dodać publicznego adresu do prywatnego IP, który należy do agenta.

Stworzenie private endpoint w FW w acr do tej podsieci nic nie zmienia.

Jakieś pomysły jak to zrobić?

**Comment by mgrabarz** [comment]
Powinno dać radę na wiele sposobów:

1. ACR premium za private linkiem, który to private link jest wdrożony do sieci lub sieci peer z agentami.. współdzielą obie dnsa.
2. ACR z firewallem na service endpoint. Service endpoint wdrożony w podsieci z agentami.
3 AcR z publicznym adresem IP i firewallem. Maszyna wirtualna z agentem wychodzi statycznym adresem IP, takie coś może być ogarnięte na kilka sposobów:
- za pomocą Nat Gateway
- za pomocą Azure Firewall z forced tunellingiem (przez UDR lub Azure Route Service)
- podobnie jak wyżej, ale zamiast firewalla jest urządzenie NVA z marketplace
- reverse proxy przez Application Gateway.

**Comment by piotrgwiazda** [comment]
Ten ACR Task?

https://docs.microsoft.com/en-us/azure/container-registry/container-registry-tasks-overview




pon., 7 mar 2022, 18:04 użytkownik Marek Grabarz ***@***.***>
napisał:

**Reply by mgrabarz** [reply]
Założyłem, że nie chodzi o TE acr taski ;)

**Reply by cybul7** [reply]
Próbowałem jeszcze zrobić NAT gateway dla subnetu w którym jest ten acr agent pool.
Niestety nadal strzela do ACR z losowych publicznych adresów ip.

**Reply by mgrabarz** [reply]
Co to jest ACR agent pool? To są VMki agentów Azure DevOps, VMSS? Czy jesteś pewien że odpalasz pipeline na tej puli?

**Reply by cybul7** [reply]
Chyba o czymś innym rozmawiamy.
Tutaj masz screen z ACR:
Dostępne w zakładce ACR -> Services -> Tasks -> Agent pools
![image](https://user-images.githubusercontent.com/56821086/157285607-473b2df9-7352-41e3-ac3f-da68ef32eeeb.png)

**Reply by mgrabarz** [reply]
Wychodzi na to, że używasz ACR taskow,  a nie ADO.

W takim razie zapomnij o wszystkim co pisałem wyżej. Jedyna opcja to whitelist całego zakresu IP dla ServiceTag AzureContainerRegistry.<Region> na firewallu. https://docs.microsoft.com/en-us/azure/container-registry/container-registry-firewall-access-rules#allow-access-by-ip-address-range

**Reply by cybul7** [reply]
O to właśnie chodzi.
Ta lista załatwia temat.
Dodam ją jako wyjątki w FW i będzie ok. 
Dzięki.

**Reply by cybul7** [reply]
Niestety jest problem:
The requested feature IP rule exceeds allowed quota 100 for the SKU Premium.

**Reply by mgrabarz** [reply]
Wsadzisz wszystkie ACR IP na firewall, czy tylko z AzureContainerRegistry.WestEurope ( czy jaki tam używasz)?

Serio jest tam ponad 100 CIDRów? Wierzyć się nie chce.

**Comment by huczas** [comment]
a nie prościej ustawić retention policy w acr np na 1 dzień?

**Reply by cybul7** [reply]
Nie jest to możliwe w tym przypadku.
Nie wszyskie obrazy będą usuwane, w tasku jest odpowiedni filtr.
